backend/src/tests/Equinor.ProjectExecutionPortal.Tests.WebApi/IntegrationTests/OnboardedAppControllerTests.cs

```diff
@@ -0,0 +1,199 @@
+            await Assert.ThrowsExceptionAsync<InvalidActionException>(() => AddOnboardedApp(UserType.Authenticated, addOnboardedAppPayload));
+        }
+
+        [TestMethod]
+        public async Task Remove_OnboardedApp_AsAuthenticatedUser_ShouldReturnOk()
+        {
+            // Arrange
+            var addOnboardedAppPayload = new ApiOnboardAppRequest
+            {
+                AppKey = "app-to-be-removed"
+            };
+
+            // Act
+            await AddOnboardedApp(UserType.Authenticated, addOnboardedAppPayload);
+
+            var getAllAfterAddedResponse = await GetAllOnboardedApps(UserType.Authenticated);
+            var getAllAfterAddedContent = await getAllAfterAddedResponse.Content.ReadAsStringAsync();
+            var totalCountAfterAdded = JsonConvert.DeserializeObject<IList<ApiOnboardedApp>>(getAllAfterAddedContent)?.Count;
+
+            var removeResponse = await RemoveOnboardedApp(UserType.Authenticated, addOnboardedAppPayload.AppKey);
+
+            var getAllAfterRemovalRepsonse = await GetAllOnboardedApps(UserType.Authenticated);
+            var getAllAfterRemovalContent = await getAllAfterRemovalRepsonse.Content.ReadAsStringAsync();
+            var totalCountAfterRemoval = JsonConvert.DeserializeObject<IList<ApiOnboardedApp>>(getAllAfterRemovalContent)?.Count;
+
+            // Assert
+            if (totalCountAfterRemoval != null && totalCountAfterAdded != null)
+            {
+                Assert.AreEqual(removeResponse.StatusCode, HttpStatusCode.NoContent);
+                Assert.AreEqual(totalCountAfterRemoval, totalCountAfterAdded - 1);
+            }
+            else
+            {
+                Assert.Fail();
+            }
+        }
+
+        [TestMethod]
+        public async Task Remove_OnboardedApp_AsAnonymousUser_ShouldReturnUnauthorized()
+        {
+            // Arrange
+            var existingOnboardedAppKey = OnboardedAppsData.InitialSeedData.OrgChartApp.AppKey;
+
+            // Act
+            var removeResponse = await RemoveOnboardedApp(UserType.Anonymous, existingOnboardedAppKey);
+
+            // Assert
+            Assert.AreEqual(removeResponse.StatusCode, HttpStatusCode.Unauthorized);
+        }
+
+        private static async Task<HttpResponseMessage> AddOnboardedApp(UserType userType, ApiOnboardAppRequest onboardApp)
+        {
+            var serializePayload = JsonConvert.SerializeObject(onboardApp);
+            var content = new StringContent(serializePayload, Encoding.UTF8, "application/json");
+
+            var client = TestFactory.Instance.GetHttpClient(userType);
+            var response = await client.PostAsync($"{Route}", content);
+
+            return response;
+        }
+
+        private static async Task<HttpResponseMessage> GetAllOnboardedApps(UserType userType)
+        {
+            var client = TestFactory.Instance.GetHttpClient(userType);
+            var response = await client.GetAsync($"{Route}");
+
+            return response;
```

Comment:
Deserialize in return instead of deserializing when used

Reply:
Will have to perform certain assertions in that method as well then. I created a method called "AssertGetAllOnboardedApps" that performs some assertions and condition logic and replaced the individual deserializations with that.